packages/api/lambdas/index-from-database.js

```diff
@@ -98,11 +116,83 @@ async function indexModel({
   return totalItemsIndexed;
 }
 
+/**
+* indexModel - Index a postgres RDS table's contents to ElasticSearch
+*
+* @param {Object} params                  -- parameters
+* @param {any} params.esClient            -- ElasticSearch client
+* @param {any} params.postgresModel       -- @cumulus/db model
+* @param {string} params.esIndex          -- esIndex to write records to
+* @param {any} params.indexFn             -- Indexer function that maps to the database model
+* @param {any} params.limitEsRequests     -- limitEsRequests method (used for testing)
+* @param {Knex} params.knex               -- configured knex instance
+* @param {any} params.translationFunction -- function to translate postgres record
+*                                            to API record for ES
+* @param {number} params.pageSize         -- Page size for postgres pagination
+* @returns {number}                       -- number of items indexed
+*/
+async function indexModel({
+  esClient,
+  postgresModel,
+  esIndex,
+  indexFn,
+  limitEsRequests,
+  knex,
+  translationFunction,
+  pageSize,
+}) {
+  let startId = 1;
+  let totalItemsIndexed = 0;
+  let done;
+  let maxIndex = await postgresModel.getMaxCumulusId(knex);
+  /* eslint-disable no-await-in-loop */
+  while (done !== true && maxIndex > 0) {
+    const pageResults = await postgresModel.paginateByCumulusId(knex, startId, pageSize);
+    log.info(
+      `Attempting to index ${pageResults.length} records from ${postgresModel.tableName}`
+    );
+
+    const translatedResults = await Promise.all(
+      pageResults.map(async (result) => await translationFunction(result))
+    );
+
+    const indexPromises = translatedResults.map((result) => limitEsRequests(
+      async () => {
+        try {
+          return await indexFn(esClient, result, esIndex);
```

Comment:
Why not do the translation and indexing in the same callback? Doing Promise.all twice over the items seems unnecessary

Reply:
Also the `Promise.all` for translation doesn't have its concurrency limited, but I think it would be preferrable if it did